### ISSUE TITLE
Compile Error mutations should never be reused from previous result

### ISSUE THREAD

**Issue by Mobrockers** [stryker-mutator/stryker-net#1295]
**Describe the bug**
#1290 happens because compile error mutations will never have covering tests as these mutations are rolled back. Aside from the NPE on coveringTests being null when there are no coveringTests, compile error mutation results should never be reused in the first place as in the best case they'll be rolled back again anyway so checking them doesn't add anything and in the worst case it marks mutations as compile error that might not be a compile error anymore.

We should ignore any mutation result that is a compile error, it should never be used in dashboard compare.

**Comment by vanashimko** [comment]
I started looking into that too and have a question. In `DiffMutantFilter` when running with dashboard compare enabled there is a `SetMutantStatusToBaselineMutantStatus` which sets `NotRun` status with some "inconclusive" reason (seems to be this is not covered by unit tests and I did not understand the background): https://github.com/stryker-mutator/stryker-net/blob/7d4d0231a9de4d137133d45814d9418cdff0ca73/src/Stryker.Core/Stryker.Core/MutantFilters/DiffMutantFilter.cs#L154-L158

I tried to handle compile error mutants here and just set `NotRun` for all matching mutants if baseline one has compile error status.

And it did not work because of `SetNotRunMutantsToIgnored` call in https://github.com/stryker-mutator/stryker-net/blob/7d4d0231a9de4d137133d45814d9418cdff0ca73/src/Stryker.Core/Stryker.Core/MutantFilters/DiffMutantFilter.cs#L95-L103

So basically all `NotRun` set when working with baseline analysis (I assume `NotRun` is set to run them) are converted to `Ignored` and not run actually. Again I may be missing the bigger picture and I did not get what this "inconclusive" means, so can not get some reproducible example, but from the "static code analysis" standpoint it looks strange.

**Comment by Mobrockers** [comment]
There can be multiple mutants covering the exact same span in the file (== mutated to >= but also <= for example) and if we cannot determine which is which (more than one matchingMutant) we have to set the mutants to NotRun with the reason 'inconclusive'. Basically we have to test them because we cannot determine which result from the baseline would fit.

> I tried to handle compile error mutants here and just set NotRun for all matching mutants if baseline one has compile error status.

You bring up a good point and I have actually been wondering if the dashboard compare filter should simply be separated from the DiffMutantFilter. Someone could turn on both if wanted but perhaps someone might want to only use dashboard compare (where we do use git information to reset changed mutants to NotRun) and don't set unchanged mutants to Ignored.

**Comment by vanashimko** [comment]
Thanks for the explanation!

> the mutation process that runs before the filtering process
> We should simply not touch them as their final status has already been determined.

Does it mean that we can ignore compile error mutants in _all_ filters at all?

For example do `.Where(x => x.ResultStatus != MutantStatus.CompileError)` here
https://github.com/stryker-mutator/stryker-net/blob/7d4d0231a9de4d137133d45814d9418cdff0ca73/src/Stryker.Core/Stryker.Core/MutationTest/MutationProcess.cs#L128
of here https://github.com/stryker-mutator/stryker-net/blob/7d4d0231a9de4d137133d45814d9418cdff0ca73/src/Stryker.Core/Stryker.Core/MutantFilters/BroadcastMutantFilter.cs#L22

**Comment by Mobrockers** [comment]
Yes we should probably ignore them in all filters, good point!
